### ISSUE TITLE
Build fail with ERS_Ln9.ne4pg2_ne4pg2.F-MMF1.cori-knl_intel.eam-crmout

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/E3SM#4246]
Using E3SM master of Apr22nd (with a fix for location of PERL libxml):

```
ERS_Ln9.ne4pg2_ne4pg2.F-MMF1.cori-knl_intel.eam-crmout fails to build with:

/global/cscratch1/sd/ndk/wacmy/ndk_machinefiles_cori-add-PERL5LIB/components/eam/src/physics/cam/check_energy.F90(709): error #6460: This is not a field name that is defined in the encompassing structure.   [H2OTEMP]
          ienet(:ncol,lchnk) = cam_in(lchnk)%h2otemp(:ncol)
---------------------------------------------^
/global/cscratch1/sd/ndk/wacmy/ndk_machinefiles_cori-add-PERL5LIB/components/eam/src/physics/cam/check_energy.F90(709): error #6158: The structure-name is invalid or is missing.   [CAM_IN]
          ienet(:ncol,lchnk) = cam_in(lchnk)%h2otemp(:ncol)
-------------------------------^
compilation aborted for /global/cscratch1/sd/ndk/wacmy/ndk_machinefiles_cori-add-PERL5LIB/components/eam/src/physics/cam/check_energy.F90 (code 1)
```

**Comment by rljacob** [comment]
All the MMF cases are failing to build.

**Comment by whannah1** [comment]
We already know about this error and it will be fixed by PR #4145. It's simply a matter of deleting the MMF version of the surface exchange routines (src/physics/crm/camsrfexch.F90). This file was duplicated to facilitate the original implementation of MAML, which is removed by PR #4145 in favor of a new approach that is being developed. 

BTW, there's a separate restart bug that is preventing the MMF from being run with GNU and is not covered by the ERS test, but will be fixed by PR #4239.

**Comment by brhillman** [comment]
@ndkeen PR #4254 should have fixed this, and it looks like tests are passing on cori-knl now, so I am going to close this issue. Please re-open if you still have trouble. Thanks!
